ddtrace/internal/runtime/__init__.py

```diff
@@ -9,4 +12,50 @@
     "RuntimeTags",
     "RuntimeMetrics",
     "RuntimeWorker",
+    "get_runtime_id",
 ]
+
+
+def _generate_runtime_id():
+    return uuid.uuid4().hex
+
+
+_RUNTIME_ID = _generate_runtime_id()
+
+
+if hasattr(os, "register_at_fork"):
+
+    def _set_runtime_id():
+        global _RUNTIME_ID
+        _RUNTIME_ID = _generate_runtime_id()
+
+    os.register_at_fork(after_in_child=_set_runtime_id)
+
+    def get_runtime_id():
+        return _RUNTIME_ID
+
+
+else:
+    # Non-POSIX systems or Python < 3.7
+    import threading
+
+    _RUNTIME_PID = os.getpid()
+
+    def _set_runtime_id():
+        global _RUNTIME_ID, _RUNTIME_PID
+        _RUNTIME_ID = _generate_runtime_id()
+        _RUNTIME_PID = os.getpid()
+
+    _RUNTIME_LOCK = threading.Lock()
+
+    def get_runtime_id():
+        with _RUNTIME_LOCK:
+            pid = os.getpid()
```

Comment:
I think there maybe a race condition here:
* `os.getpid` is called by thread A
* `fork` is called by thread B
* `os.getpid` get called again from inside `_set_runtime_id`

And I think results will not be correct in this case.
Would it make sense to store first result of `getpid` in a variable and use it later?

Reply:
Hm, let me try to follow your scenario if I understand correctly.

If thread B forks, `os.getpid()` is called from the child process in `_set_runtime_id` which modifies the child memory and set a new PID and UUID. If that same child calls `get_runtime_id()`, it'll get its new runtime id.

Nothing changes in the parent process.

Feel free to correct my scenario if I misread yours!

Reply:
Well, I assumed that fork may happen between `_RUNTIME_ID = _generate_runtime_id()` and `_RUNTIME_PID = os.getpid()` - and if that's not the case, I guess it is fine.